### ISSUE TITLE
Only two levels of nesting are being returned

### ISSUE THREAD

**Issue by monolithed** [strapi/strapi#18882]
### Required System information

- Node.js version: v20.9.0
- NPM version: 10.1.0
- Strapi version: 4.15.4
- Database: sqlite
- Operating system: macOS Sonoma, 14
- Is your project Javascript or Typescript: Javascript

### Describe the bug

In my data structure, there are four levels of nesting specified, but only two are being returned.

https://github.com/strapi/strapi/assets/803674/152cf780-b402-4049-8647-438510ecd185

**Request:**

```js
const data = await got.get('http://localhost:1337/api/courses?locale=ru&populate=*', {
        headers: {
            Authorization:
                'Bearer b497e21fd3e11957384e82086ec4b3a2e2c60dd8165ae57b7071446a492203945ad9baf2cff12b3f4801bfe2b940f059245e1f959fbac413e9edcabcfdb3376c3ee5acba47444cca85e01eeb063d8fc465083dbc178f795c6d139686b9daad62189315ba7640bbe12310805d70879f9c6679b8b377491acf7e9a681a8b5d19dr',
        }
}).json();
```

**Response:**

```json
{
    "status": 200,
    "body": [
        {
            "data": [
                {
                    "id": 2,
                    "attributes": {
                        "title": "Course (1)",
                        "description": "1",
                        "createdAt": "2023-11-22T20:34:07.662Z",
                        "updatedAt": "2023-11-22T21:04:34.291Z",
                        "publishedAt": "2023-11-22T20:36:03.008Z",
                        "locale": "ru",
                        "modules": {
                            "data": [
                                {
                                    "id": 2,
                                    "attributes": {
                                        "title": "Module (2)",
                                        "description": "Охота",
                                        "createdAt": "2023-11-22T20:34:49.212Z",
                                        "updatedAt": "2023-11-22T21:04:43.479Z",
                                        "publishedAt": "2023-11-22T20:37:15.024Z",
                                        "locale": "ru"
                                    }
                                }
                            ]
                        },
                        "localizations": {
                            "data": [
                                {
                                    "id": 1,
                                    "attributes": {
                                        "title": "Course",
                                        "description": "All day testing",
                                        "createdAt": "2023-11-22T18:34:14.658Z",
                                        "updatedAt": "2023-11-22T21:03:00.811Z",
                                        "publishedAt": "2023-11-22T19:41:36.159Z",
                                        "locale": "en"
                                    }
                                }
                            ]
                        }
                    }
                }
            ],
            "meta": {
                "pagination": {
                    "page": 1,
                    "pageSize": 25,
                    "pageCount": 1,
                    "total": 1
                }
            }
        }
    ]
}
```

### Actual:

Couse (1) -> Module (2):

### Expected:

Couse (1) -> Module (2) -> Lesson (3) ->  Topic (4):

**Comment by monolithed** [comment]
I apologize, but after only 2 days of using this tool, I've raised 5 tickets, spent a bunch of time investigating issues, reading documentation and code. I've started the project from scratch twice because there were bugs in the TypeScript version. And now you're sending me to the internet to find the reasons for another system behavior that looks more like a bug. If it's not a bug and people are endlessly opening tickets on this topic, maybe provide some hints in the interface so that developers don't waste their time? At first glance, this system looks fantastic, but in use, it turns life into hell.
